### ISSUE TITLE
SC1073 when capturing stderr in variable

### ISSUE THREAD

**Issue by austinbutler** [koalaman/shellcheck#698]
I have a working Bash script with a line that captures the stderr to a variable, but ShellCheck seems to have a really tough time with this method. See below.

For context, I didn't want stderr printing to screen, but I also didn't want to throw it away (dev/null) nor write out to a file if I didn't need to. This method puts the stderror into the variable without printing to the terminal.

```
$ shellcheck myscript
 
Line 1:
RESULT=$((mysql -u "$DB_USER" -p"$DB_PASS" -h "$db" -e"quit") 2>&1)
       ^-- SC1073: Couldn't parse this $((..)) expression.
       ^-- SC1009: The mentioned parser error was in this $((..)) expression.
                   ^-- SC1072: Fix any mentioned problems and try again.
```

**Comment by koalaman** [comment]
ShellCheck does indeed have some issues disambiguating cases like this.

As a temporary workaround, you can add a space and write `$( (` so ShellCheck realizes this is command substitution with a nested subshell and not a `$((..))` arithmetic expression:

```
RESULT=$( (mysql -u "$DB_USER" -p"$DB_PASS" -h "$db" -e"quit") 2>&1)
```

Also note that the subshell isn't actually necessary, it's equivalent to just this:

```
RESULT=$(mysql -u "$DB_USER" -p"$DB_PASS" -h "$db" -e"quit" 2>&1)
```

**Comment by koalaman** [comment]
This was fixed in e3bef9dc in July but I forgot to update this bug. Thanks!

The fix is actually a warning that suggests adding a space as described. This is more readable and better supported across shells.
